tests/test_send_recv_obj.py

```diff
@@ -49,12 +56,17 @@
 
 
 @pytest.mark.asyncio
-async def test_send_recv_numpy():
+@pytest.mark.parametrize("size", msg_sizes)
+async def test_send_recv_numpy(size):
     np = pytest.importorskip('numpy')
-    async with echo_pair() as (_, client):
-        msg = np.frombuffer(memoryview(b"hi"), dtype='u1')
+    x = "a"
+    x = x * size
+    msg = bytes(x, encoding='utf-8')
+    msg = memoryview(msg)
+    msg = np.frombuffer(msg, dtype='u1')
```

Comment:
This should work fine. Would also work if this were the following.

```python
msg = np.asarray(msg)
```

The benefit of this expression is it is able to infer the type from the `memoryview`. Also if the data has a shape to it (like it is a 2-D matrix), then that will be preserved. `np.frombuffer` flattens everything to a 1-D array instead.

Reply:
There's a few other lines below that could benefit from this same pattern. Not critical though.